src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -1315,23 +1304,18 @@ static ucs_status_t uct_ib_mlx5dv_md_open(struct ibv_device *ibv_device,
         goto err;
     }
 
-    md = ucs_calloc(1, sizeof(*md), "ib_mlx5_md");
+    md = ucs_derived_of(uct_ib_md_alloc(sizeof(*md), "ib_mlx5dv_md", ctx),
+                        uct_ib_mlx5_md_t);
     if (md == NULL) {
         status = UCS_ERR_NO_MEMORY;
         goto err_free_context;
     }
 
-    dev              = &md->super.dev;
-    dev->ibv_context = ctx;
+    dev = &md->super.dev;
 
     status = uct_ib_device_query(dev, ibv_device);
     if (status != UCS_OK) {
-        goto err_free;
-    }
-
-    if (!(uct_ib_device_spec(dev)->flags & UCT_IB_DEVICE_FLAG_MLX5_PRM)) {
```

Comment:
why no need to check this anymore?

Reply:
@Artemy-Mellanox i didn't notice that we're setting another `flags` from device, not from device spec.
I guess this check should be here

Reply:
@yosefe @dmitrygx in case we have mlx5 support compiled we detect mlx5 supporting device with `mlx5dv_is_supported()`

Reply:
so this check is not needed, but it's not really part of fixing leak?

Reply:
@yosefe didn't see your comment when pushed changes suggested by @Artemy-Mellanox